examples/webview2/src/main.rs

```diff
@@ -0,0 +1,622 @@
+use std::{
+    collections::HashMap,
+    ffi::CString,
+    mem,
+    sync::{mpsc, Arc, Mutex},
+};
+
+use bindings::{
+    Microsoft::Web::WebView2::Core::*,
+    Windows::{
+        Foundation::*,
+        Win32::{
+            self,
+            DisplayDevices::{RECT, SIZE},
+            Gdi,
+            HiDpi::{self, PROCESS_DPI_AWARENESS},
+            KeyboardAndMouseInput,
+            MenusAndResources::HMENU,
+            SystemServices::{self, HINSTANCE, LRESULT, PSTR},
+            WindowsAndMessaging::{
+                self, SetWindowPos_uFlags, HWND, LPARAM, MSG, SHOW_WINDOW_CMD, WINDOW_EX_STYLE,
+                WINDOW_LONG_PTR_INDEX, WINDOW_STYLE, WNDCLASSA, WPARAM,
+            },
+        },
+    },
+};
+
+use windows::*;
+
+use serde::Deserialize;
+use serde_json::{Number, Value};
+
+fn main() -> Result<()> {
+    initialize_sta()?;
+    set_process_dpi_awareness()?;
+
+    let webview = WebView::create(true, None)?;
+
+    // Bind a quick and dirty calculator callback.
+    let bound = webview.clone();
+    webview.bind("hostCallback", move |id, request| {
+        if request.len() == 3 {
+            if let Some(Value::String(str)) = request.get(0) {
+                if str == "Add" {
+                    if let Some(Value::Number(a)) = request.get(1) {
+                        if let Some(Value::Number(b)) = request.get(2) {
+                            let result = a.as_f64().unwrap_or(0f64) + b.as_f64().unwrap_or(0f64);
+                            let result = Number::from_f64(result);
+                            if let Some(result) = result {
+                                let result = Value::Number(result);
+                                bound.resolve(id, 0, result).unwrap_or(&bound);
```

Comment:
Is the `unwrap_or` here what you want? This just unwraps the result and ignores it.

Reply:
Still new to Rust (this experiment was actually my first project not from the Rust Book). What I really want is the `?` operator, but I was running into trouble with mapping different Error types to the Result. This was me getting tired of handling that and deciding to just ignore the error (since it takes arbitrary input from a call on the JS console, I decided crashing with `unwrap` would be unfriendly).

One way to deal with that which I'm thinking about trying is to make my own Error enum which can wrap all of the Error types I expect, and implement From for those types. I found at least one blog which ends up suggesting that, did I just reinvent another wheel/idiom? 😆